### ISSUE TITLE
Node.js + Nuxt.js  contentWindow.widgetReady is not a function

### ISSUE THREAD

**Issue by awesner** [tradingview/charting-library-examples#77]
Node.js + Nuxt.js

I added 'charting_library.min.js' file to the 'static' folder in my nuxt.js folder and connect this file inside my nuxt.config.js file like this:
head: {
script: [
      {
        src: '/js/charting-library.min.js',
      }
  ]
}
After that I downloaded TVContainer.vue example component from you github and added it to my project. But when I run the page I get this error: "Uncaught TypeError: i.contentWindow.widgetReady is not a function
    at HTMLIFrameElement.n (charting-library.min.js:1)"
Could you help me out, please?

**Comment by timocov** [comment]
We don't have example with nuxt.js. Where did you get it?

**Comment by awesner** [comment]
Hi Evgeniy,

we used this sample  https://github.com/tradingview/charting-library-examples/tree/master/vuejs and added in vue+nuxt client.

Uncaught TypeError: i.contentWindow.widgetReady is not a function at HTMLIFrameElement.n (charting-library.min.js
Start: Frontend_Node+nuxt_with_tradingview_in_trading_page_issue
Npm run dev  User: root PW: 1234

Thx,
 Skype: alexander_wesner
 Issue
I added 'charting_library.min.js' file to the 'static' folder in my nuxt.js folder and connect this file inside my nuxt.config.js file like this:
head: {
script: [
      {
        src: '/js/charting-library.min.js',
      }
  ]
}
After that I downloaded TVContainer.vue example component from you github and added it to my project. But when I run the page I get this error: "Uncaught TypeError: i.contentWindow.widgetReady is not a function
    at HTMLIFrameElement.n (charting-library.min.js


Von: Evgeniy Timokhov <notifications@github.com>
Gesendet: Dienstag, 15. Januar 2019 12:09
An: tradingview/charting-library-examples <charting-library-examples@noreply.github.com>
Cc: awesner <alexander.wesner@outlook.com>; Author <author@noreply.github.com>
Betreff: Re: [tradingview/charting-library-examples] Node.js + Nuxt.js contentWindow.widgetReady is not a function (#77)


We don't have example with nuxt.js. Where did you get it?

**Comment by timocov** [comment]
We aren't familiar with this framework and unfortunately cannot help you. But if you figure out what's going on - please write a comment how did you solve it. It might help someone who faces the same problem.

**Comment by awesner** [comment]
Hi,

I thing that is not a Framework Problem, maybe the Libery is in ES6 compiled.

Do you know if exist tradingview Libery in ES6?

Thx,
Alexander

Von: Evgeniy Timokhov <notifications@github.com>
Gesendet: Dienstag, 15. Januar 2019 13:03
An: tradingview/charting-library-examples <charting-library-examples@noreply.github.com>
Cc: awesner <alexander.wesner@outlook.com>; Author <author@noreply.github.com>
Betreff: Re: [tradingview/charting-library-examples] Node.js + Nuxt.js contentWindow.widgetReady is not a function (#77)


We aren't familiar with this framework and unfortunately cannot help you. But if you figure out what's going on - please write a comment how did you solve it. It might help someone who faces the same problem.

**Comment by rainrain002** [comment]
I have same problem. please share if you already fixed it.

**Comment by timocov** [comment]
Guys, can you at least share the code (please be sure that the library's code is not public accessible in your code/example) and steps to achieve that error? I'm sure that the problem is in the configuration, not in the library itself.

**Comment by rainrain001** [comment]
Hi. i also encounter this problem. i tried to change the code just like this
from -> `i.contentWindow.widgetReady( function() ....` to -> `i.contentWindow.widgetReady = function()` and it works. but when i tried to put back the original code the error will fix already. i don't know why. but it works for me.

**Comment by timocov** [comment]
> and it works

This is incorrect code. The original code calls a function inside iframe, but you just override it.

The error `contentWindow.widgetReady is not a function` might be happened if inner iframe does not load properly (404, invalid url or so on). Please check it.

**Comment by poruchik85** [comment]
Hi, guys.

I spent a couple of days connecting the library to the nuxt, and this is how I did it:

As said **awesner**, in `nuxt.config.js` insert into the `head` section 
```javascript
script: [
	{ src: '/charting_library/charting_library.min.js', type: 'text/javascript'},
	{ src: '/datafeeds/udf/dist/polyfills.js', type: 'text/javascript'},
	{ src: '/datafeeds/udf/dist/bundle.js', type: 'text/javascript'},
]
```

datafeeds are needed only for testing, after implementing your own datafeeds, you can remove them from here.

Then I connected the TVContainer.vue component from the example as a plugin:

`client/plugins/trading-view.js:`
```javascript
import Vue from 'vue';
import TVChartContainer from '~/components/trading/TVChartContainer';

Vue.component('tv-chart-container', TVChartContainer);
```

And modified, removing the charting_library.min.js from it (it is already connected on the page) and changing
`const tvWidget = new widget(widgetOptions);`
to
`const tvWidget = new window.TradingView.widget(widgetOptions);`

In `nuxt.config.js` connected a plugin:
```javascript
plugins: [
	...
    { src: '~/plugins/trading-view', ssr: false },
	...
],
```

Most importantly, the resources (the charting_library and the datafeeds folders from the library) should be in your static folder and be accessible by direct URL - this is what **timocov** mentioned in the last message.

**Comment by fasterv410** [comment]
> Hi, guys.
> 
> I spent a couple of days connecting the library to the nuxt, and this is how I did it:
> 
> As said **awesner**, in `nuxt.config.js` insert into the `head` section
> 
> ```js
> script: [
> 	{ src: '/charting_library/charting_library.min.js', type: 'text/javascript'},
> 	{ src: '/datafeeds/udf/dist/polyfills.js', type: 'text/javascript'},
> 	{ src: '/datafeeds/udf/dist/bundle.js', type: 'text/javascript'},
> ]
> ```
> 
> datafeeds are needed only for testing, after implementing your own datafeeds, you can remove them from here.
> 
> Then I connected the TVContainer.vue component from the example as a plugin:
> 
> `client/plugins/trading-view.js:`
> 
> ```js
> import Vue from 'vue';
> import TVChartContainer from '~/components/trading/TVChartContainer';
> 
> Vue.component('tv-chart-container', TVChartContainer);
> ```
> 
> And modified, removing the charting_library.min.js from it (it is already connected on the page) and changing
> `const tvWidget = new widget(widgetOptions);`
> to
> `const tvWidget = new window.TradingView.widget(widgetOptions);`
> 
> In `nuxt.config.js` connected a plugin:
> 
> ```js
> plugins: [
> 	...
>     { src: '~/plugins/trading-view', ssr: false },
> 	...
> ],
> ```
> 
> Most importantly, the resources (the charting_library and the datafeeds folders from the library) should be in your static folder and be accessible by direct URL - this is what **timocov** mentioned in the last message.

It Working, Thank you.

**Comment by learntolearn99** [comment]
Hi @fasterv410 and @poruchik85 
I'm getting this error
in ./components/exchange/TVChartContainer.vue?vue&type=script&lang=js&

"export 'widget' was not found in '../../assets/js/charting_library.min.js'

**Comment by learntolearn99** [comment]
Hi
who anyone can help me remove  white border in tradingview ?
![image](https://user-images.githubusercontent.com/54758734/65036104-e1d86600-d974-11e9-918d-70eaeeecd209.png)

**Comment by timocov** [comment]
@learntolearn99 see https://github.com/tradingview/charting_library/issues/4320#issuecomment-532145366
